azure-kusto-ingest/azure/kusto/ingest/ingest_client.py

```diff
@@ -108,3 +127,37 @@ def ingest_from_blob(self, blob_descriptor: BlobDescriptor, ingestion_properties
         content = ingestion_blob_info_json
         queue_client = queue_service.get_queue_client(queue=random_queue.object_name, message_encode_policy=TextBase64EncodePolicy())
         queue_client.send_message(content=content)
+
+    def _validate_endpoint_service_type(self):
+        if self._endpoint_service_type is None or not self._endpoint_service_type.strip():
```

Comment:
Maybe my question was lost in this long discussion, so asking again: is it reasonable to get a whitespace here? 
I not, simplify this check (and all other places) to "if self._endpoint_service_type"

Reply:
Yeah my response got buried, sorry about that. In https://github.com/Azure/azure-kusto-python/pull/267#discussion_r488602561 I wrote:
My coding style places an emphasis on resilience. Right now, it can't have whitespace, but perhaps it could in the future. This method is a black box that should produce the correct output given any reasonable input. It's similar to (I presume) why in our C# code base we prefer to use "string.IsNullOrWhiteSpace()" wherever possible.

Reply:
A black box defined by its input/output - so if the "contract" of the method is to return correct endpoint or empty 
 or None it is it's responsibility to return the right result so whoever use this method won't have to check unexpected values everywhere. 
You can verify in the method if you are concerned about getting whitespace.

Reply:
Exactly, except inputs can also be inputs from other method calls within the method, and not necessarily just its literal signature.

"You can verify in the method if you are concerned about getting whitespace."
That's what I do here.
I'm not sure I followed your point, because the way I'm reading it, it seems you agree with me.